fdbserver/workloads/ExcludeIncludeStorageServersWorkload.actor.cpp

```diff
@@ -0,0 +1,153 @@
+// 		2) then verify RateKeeper is only bookkeeping required processes. (delete the SS that has quitted)
+// it's okay to quit if we can't create such scenario in rare cases, as long as in most cases we can create the scenario
+// we have seen the case RK bookkeeping large number of unnecessary SS:
+// https://github.com/apple/foundationdb/issues/10260
+struct ExcludeIncludeStorageServersWorkload : TestWorkload {
+	static constexpr auto NAME = "ExcludeIncludeStorageServers";
+	bool enabled;
+
+	std::map<AddressExclusion, std::set<AddressExclusion>> machineProcesses; // ip -> ip:port
+
+	ExcludeIncludeStorageServersWorkload(WorkloadContext const& wcx) : TestWorkload(wcx) {
+		enabled =
+		    !clientId && g_network->isSimulated(); // only do this on the "first" client, and only when in simulation
+		if (g_network->isSimulated()) {
+			g_simulator->allowLogSetKills = false;
+		}
+	}
+
+	void disableFailureInjectionWorkloads(std::set<std::string>& out) const override {
+		// Failure injection workloads like Rollback, Attrition and so on are interfering with the test.
+		// In particular, this test expects the storage server can be excluded and included in time
+		// to test that ratekeeper's bookkeeping works fine.
+		// Consequently, we disable all failure injection workloads in backgroud for this test
+		out.insert("all");
+	}
+
+	Future<Void> setup(Database const& cx) override { return Void(); }
+
+	Future<Void> start(Database const& cx) override {
+		if (!enabled)
+			return Void();
+		return workloadMain(this, cx);
+	}
+
+	Future<bool> check(Database const& cx) override { return true; }
+
+	void getMetrics(std::vector<PerfMetric>&) override {}
+
+	ACTOR static Future<Void> workloadMain(ExcludeIncludeStorageServersWorkload* self, Database cx) {
+		state int round = deterministicRandom()->randomInt(10, 80);
+		state std::set<AddressExclusion> servers;
+		state std::set<AddressExclusion>::iterator it;
+		state Transaction tr(cx);
+		state double timeout = 100.0;
+		state int timeoutCnt = 0;
+		state int maxTimeout = std::min(30, round);
+		TraceEvent("WorkloadStart").detail("Round", round).log();
+		loop {
+			if (timeoutCnt > maxTimeout) {
+				TraceEvent("QuitEarlyNotCompleteServerExclude").log();
+				break;
+			}
+			try {
+				servers.clear();
+				// including an invalid address means include everything(clear all exclude prefix)
+				wait(includeServers(cx, std::vector<AddressExclusion>(1)));
+
+				tr = Transaction(cx);
+				tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+				tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+				tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+
+				// get all storage servers
+				std::vector<std::pair<StorageServerInterface, ProcessClass>> results =
+				    wait(NativeAPI::getServerListAndProcessClasses(&tr));
+				for (auto& [ssi, p] : results) {
+					servers.insert(AddressExclusion(ssi.address().ip, ssi.address().port));
```

Comment:
It's better to skip protected addresses, e.g., coordinators. This can be checked like
```
    if (g_simulator->protectedAddresses.count(ssi.address()) == 0) 
        servers.insert(AddressExclusion(ssi.address().ip, ssi.address().port));
```

Reply:
will do